scripts/generate-google3-docs/resource-reference/main.go

```diff
@@ -293,13 +305,25 @@ func constructResourceForGVK(gvk schema.GroupVersionKind, smLoader *servicemappi
 		}
 	} else {
 		if err := handleAnnotationsAndIAMSettingsForTFBasedResource(r, gvk, smLoader); err != nil {
-			return nil, fmt.Errorf("error processing the TF based resource %v: %w", gvk, err)
+			// TODO: Add annotation and IAM settings handling logic for direct GKs.
+			if isDirectGVK &&
+				strings.Contains(err.Error(), fmt.Sprintf("unable to get service mapping: no mapping with name '%s' found", gvk.Group)) {
+				log.Printf("reference doc for direct GK '%v' doesn't cover annotations and IAM settings", gvk.GroupKind().String())
+			} else {
+				return nil, fmt.Errorf("error processing the TF based resource %v: %w", gvk, err)
+			}
 		}
 	}
 
 	r.SampleYamls, err = buildSamples(r.Kind, sampleDirPathForGVK(gvk), smLoader)
 	if err != nil {
-		return nil, fmt.Errorf("error building samples: %w", err)
+		// TODO: Samples should also be required for direct CRDs.
+		if isDirectGVK &&
+			strings.Contains(err.Error(), fmt.Sprintf("k8s-config-connector/config/samples/resources/%s: no such file or directory", strings.ToLower(gvk.Kind))) {
```

Comment:
I like this check, but I think this could fail on renamed repo.

Reply:
By renamed repo, did you mean that we might rename our repo from `k8s-config-connector` to something else?

I consider this check temporary because samples are actually pretty important and we should fill the sample gap asap. I'd assume this check will be removed in early Q4. Based on this timeline, does it still matter?

Reply:
no, I mean people could git clone the repo to a different local directory, in which case this code won't work. (I recalled someone named their repo `kcc` rather than k8s-config-connector).

Reply:
Gotcha! Removed `k8s-config-connector` in the error message.

Reply:
nite: maybe we can construct the matching string using "repo.GetResourcesSamplesPath()" and gvk?